src/main/java/com/minecolonies/entity/ai/citizen/farmer/EntityAIWorkFarmer.java

```diff
@@ -21,520 +21,550 @@
+      *
+      * @param position the position to check.
+      * @param field    the field close to this position.
+      * @return true if should be hoed.
+      */
+     private boolean shouldPlant(@NotNull BlockPos position, @NotNull Field field)
+     {
+         @Nullable ItemStack itemStack = BlockUtils.getItemStackFromBlockState(world.getBlockState(position.up()));
+
+         if (itemStack != null && itemStack.getItem() == field.getSeed())
+         {
+             requestSeeds = false;
+         }
+
+         return !field.isNoPartOfField(world, position) && !(world.getBlockState(position.up()).getBlock() instanceof BlockCrops)
+                 && !(world.getBlockState(position).getBlock() instanceof BlockHutField) && world.getBlockState(position).getBlock() == Blocks.FARMLAND;
+     }
+
+     /**
+      * Plants the crop at a given location.
+      *
+      * @param item     the crop.
+      * @param position the location.
+      */
+     private boolean plantCrop(Item item, @NotNull BlockPos position)
+     {
+         int slot = worker.findFirstSlotInInventoryWith(item);
+         if (slot != -1)
+         {
+             @NotNull IPlantable seed = (IPlantable) item;
+             world.setBlockState(position.up(), seed.getPlant(world, position));
+             getInventory().decrStackSize(slot, 1);
+             requestSeeds = false;
+             //Flag 1+2 is needed for updates
+             return true;
+         }
+         else
+         {
+             return false;
+         }
+     }
+
+     /**
+      * Terminates the planting process and resets the task.
+      *
+      * @param buildingFarmer the building of the farmer.
+      * @param field          the field being planted.
+      * @return the next state.
+      */
+     @NotNull
+     private AIState terminatePlanting(@NotNull BuildingFarmer buildingFarmer, @NotNull Field field)
+     {
+         if (requestSeeds)
+         {
+             chatSpamFilter.talkWithoutSpam("entity.farmer.NeedSeed", field.getSeed().getItemStackDisplayName(new ItemStack(field.getSeed())));
+         }
+         else
+         {
+             buildingFarmer.getCurrentField().setNeedsWork(false);
+             buildingFarmer.getCurrentField().setFieldStage(PLANTED);
+         }
+
+         resetVariables();
+         return AIState.IDLE;
+     }
+
+     /**
+      * Executes the harvesting of the field.
+      *
+      * @return the next state.
+      */
+     @NotNull
+     private AIState harvest()
+     {
+         @Nullable BuildingFarmer buildingFarmer = getOwnBuilding();
+
+         if (buildingFarmer == null || buildingFarmer.getCurrentField() == null)
+         {
+             return AIState.PREPARING;
+         }
+         @Nullable Field field = getOwnBuilding().getCurrentField();
+
+         if (workingOffset != null)
+         {
+             BlockPos position = field.getLocation().down().south(workingOffset.getZ()).east(workingOffset.getX());
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Local variable 'position' could be declared final [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=pmd%3ALocalVariableCouldBeFinal)